### PR TITLE
Fix/card section removing image source deprecation

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Fix/CardSection removing imageSource deprecation

## Changelog
CardSection - removing 'imageSource' deprecation

### CODE REVIEW

File: src/components/card/CardSection.tsx [hunk 1]
```diff
@@ -56,14 +54,6 @@ type Props = CardSectionProps & asCardChildProps;
...
   static displayName = 'Card.Section';
 
-  constructor(props: Props) {
-    super(props);
-
-    if (props.imageSource) {
-      LogService.deprecationWarn({component: 'CardSection', oldProp: 'imageSource', newProp: 'source'});
-    }
-  }
-
```
Comment (ethanshar) [inline src/components/card/CardSection.tsx:66]: Unfortunately, I think now we need to have the opposite warning.. 
It's been for a few months, no?

**Review by ethanshar — COMMENTED** [review]
@Inbal-Tish 
See my comment. 
Are we sure we can just remove the `source` prop? 
If we've been instructing people to use it for some time, we might need to deprecate it with a warning...

**Comment by Inbal-Tish** [comment]
@ethanshar I think we should leave it as 'source', same as in StateScreen. This is the Image prop's name and the description explains that this is the source for the image. I don't think we need to have our own prop name for it....

**Comment by ethanshar** [comment]
> @ethanshar I think we should leave it as 'source', same as in StateScreen. This is the Image prop's name and the description explains that this is the source for the image. I don't think we need to have our own prop name for it....

I disagree. 
When a component is image orientated then I think it make sense to keep the prop name as `source` (Avatar, Thumbnail..)
Where the component is not necessarily based on an image, I think it's important to explicitly call the prop `imageSource`. 

This is what we agreed on When Miki created this component, I don't want to change it because we accidentally migrated the prop.

**Comment by Inbal-Tish** [comment]
We did accidentally migrated it. In this case, StateScreen should be un-migrated as well, no?

**Pull request closed without merging**